app/src/main/java/com/gemwallet/android/ui/components/InfoBottomSheet.kt

```diff
@@ -0,0 +1,275 @@
+    onClose: () -> Unit,
+) {
+    InfoBottomSheet(
+        icon = info.icon,
+        painterSmall = when (info.transactionState) {
+            TransactionState.Pending -> painterResource(com.gemwallet.android.R.drawable.ic_tr_state_pending)
+            TransactionState.Confirmed -> painterResource(com.gemwallet.android.R.drawable.ic_tr_state_success)
+            TransactionState.Failed, TransactionState.Reverted -> painterResource(com.gemwallet.android.R.drawable.ic_tr_state_error)
+        },
+        title = when (info.transactionState) {
+            TransactionState.Pending -> stringResource(R.string.transaction_status_pending)
+            TransactionState.Confirmed -> stringResource(R.string.transaction_status_confirmed)
+            TransactionState.Failed -> stringResource(R.string.transaction_status_failed)
+            TransactionState.Reverted -> stringResource(R.string.transaction_status_reverted)
+        },
+        description = when (info.transactionState) {
+            TransactionState.Pending -> stringResource(R.string.info_transaction_pending_description)
+            TransactionState.Confirmed -> stringResource(R.string.info_transaction_success_description)
+            TransactionState.Failed, TransactionState.Reverted -> stringResource(R.string.info_transaction_error_description)
+        },
+        onAction = onAction,
+        onClose = onClose,
+    )
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+private fun InfoBottomSheet(
+    icon: String = "",
+    painter: Painter? = null,
+    painterSmall: Painter? = null,
+    title: String,
+    description: String,
+    buttonText: String = stringResource(R.string.common_learn_more),
+    onAction: () -> Unit,
+    onClose: () -> Unit,
+) {
+    val sheetState = rememberModalBottomSheetState()
+    val scope = rememberCoroutineScope()
+    ModalBottomSheet(
+        sheetState = sheetState,
+        shape = RoundedCornerShape(24.dp),
+        containerColor = MaterialTheme.colorScheme.background,
+        onDismissRequest = { scope.launch { sheetState.hide() }.invokeOnCompletion { onClose() } },
+        content = {
+            Column(
+                modifier = Modifier
+                    .fillMaxWidth(),
+                horizontalAlignment = Alignment.CenterHorizontally,
+            ) {
+                Box(
+                    modifier = Modifier.size(128.dp),
+                    contentAlignment = Alignment.BottomEnd
+                ) {
+                    if (painter != null) {
+                        Image(
+                            modifier = Modifier
+                                .size(120.dp)
+                                .clip(CircleShape),
+                            painter = painter,
+                            contentDescription = ""
+                        )
+                    } else {
+                        AsyncImage(
+                            modifier = Modifier
+                                .size(120.dp)
```

Comment:
So big. Could you try 64.dp

Reply:
this is the size in the IOS GemWallet that's why I took it like this 
but I can reduce it